shared/profile/search.js

```diff
@@ -0,0 +1,120 @@
+  navigateUp: () => void,
+  onChangeSearchText: (s: string) => void,
+  onSelectService: (s: string) => void,
+  _search: (term: string, service: SearchConstants.Service) => void,
+  searchText: string,
+  selectedService: string,
+  usernameText: string,
+}
+const mapStateToProps = ({profile: {searchResults}}: TypedState) => ({
+  ids: searchResults.toArray(),
+})
+const mapDispatchToProps = (dispatch: Dispatch, {navigateUp, onBack, onToggleSidePanel}: OwnProps) => ({
+  _clearSearchResults: id => dispatch(Creators.clearSearchResults()),
+  _onClick: username => {
+    dispatch(navigateUp())
+    dispatch(onUserClick(username))
+  },
+  _search: debounce(
+    (term: string, service) => dispatch(SearchCreators.search(term, 'profile:updateSearchResults', service)),
+    1e3
+  ),
+  onClose: () => dispatch(navigateUp()),
+})
+
+const Search = props => {
+  return (
+    <Box style={styleSearchContainer}>
+      <Box style={{...styleSearchRow}}>
+        <Box style={{...globalStyles.flexBoxRow, flexGrow: 1}}>
+          <Box style={{flexGrow: 1}}>
+            <UserInput
+              userItems={props.userItems}
+              showAddButton={props.showAddButton}
+              onRemoveUser={props.onRemoveUser}
+              onClickAddButton={props.onClickAddButton}
+              placeholder={props.placeholder}
+              usernameText={props.searchText}
+              onChangeText={props.onChangeText}
+            />
+          </Box>
+          <Icon style={{alignSelf: 'center'}} type="iconfont-close" onClick={props.onClose} />
+        </Box>
+      </Box>
+      <Box style={{...styleSearchRow, justifyContent: 'center'}}>
+        <Text style={{marginRight: globalMargins.tiny}} type="BodySmall">Filter:</Text>
+        <ServiceFilter selectedService={props.selectedService} onSelectService={props.onSelectService} />
+      </Box>
+      <Box style={{...styleSearchRow, ...globalStyles.scrollable}}>
+        <ResultsList items={props.ids} onClick={props.onClick} onShowTracker={props.onShowTracker} />
+      </Box>
+    </Box>
+  )
+}
+
+const styleSearchContainer = {
+  ...globalStyles.flexBoxColumn,
+  backgroundColor: globalColors.white,
+  borderRadius: 5,
+  boxShadow: `0 2px 5px 0 ${globalColors.black_20}`,
+  flex: 1,
+  left: 300,
+  minHeight: 400,
+  minWidth: 500,
+  padding: globalMargins.small,
+  position: 'absolute',
+  top: 10,
+  width: 500,
+  zIndex: 20,
+}
+
+const styleSearchRow = {
+  ...globalStyles.flexBoxRow,
+  alignItems: 'center',
+  padding: globalMargins.tiny,
+}
+
+export default compose(
+  connect(mapStateToProps, mapDispatchToProps),
+  withState('usernameText', '_onChangeText', ''),
+  withState('selectedService', '_onSelectService', 'Keybase'),
+  withState('searchText', 'onChangeSearchText', ''),
+  withHandlers({
+    onChangeText: (props: OwnProps & {_onSelectService: Function}) => nextText => {
+      props.onChangeSearchText(nextText)
+      props._search(nextText, props.selectedService)
```

Comment:
are you clearing the results if the nextText is an empty string? I guess in the future that will lead to suggestions, so it might be okay